include/dylib.hpp

```diff
@@ -220,24 +194,6 @@ class dylib {
         return m_handle;
     }
 
-    /**
-     *  @return true if a dynamic library is currently loaded in the object, false otherwise
-     */
-    operator bool() const noexcept {
-        return m_handle != nullptr;
-    }
-
-    /**
-     *  Close the dynamic library currently loaded in the object. 
-     *  This function will be automatically called by the class destructor
-     */
-    void close() noexcept {
-        if (m_handle) {
-            _close(m_handle);
-            m_handle = nullptr;
-        }
-    }
-
 private:
```

Comment:
Why not `protected`?

Reply:
You never know why people might inherit form your class.

Would you like an example: A child class which logs the symbols being used from the DLL and the time they were obtained, to examine the pattern of library usage, offline.

If there's no benefit to you in `private`izing members, make them protected by default.